### ISSUE TITLE
Train xgboost-v3 challenger focused on test Brier improvement

### ISSUE THREAD

**Issue by phead198708** [phead198708/BiGan#50]
背景
当前 xgboost-v2 在 test split 上有显著更高的 AUC 和 PR-AUC，但 Brier 略差于 logreg-baseline-v1（0.2033 vs 0.2012）。在 bootstrap 规则中，作为 probability-serving champion，这个 Brier 回退被视为硬性 caution，即使 backtest net PnL 明显更好。

目标
训练一个新的 xgboost-v3 challenger，使得在保持 test AUC 优于 baseline 的前提下，test Brier 明确优于 baseline，以降低晋升首任 champion 时的概率风险。

任务

1. 设计 xgboost-v3 配置
- 在 modeling/xgboost_v1.py 基础上新增一个 train_xgboost_v3 入口。
- v3 搜索空间相对 v2 更保守：
  - max_depth: (3, 4)
  - rounds: (100, 150, 200)
  - lambda: (5, 10) 或更高
  - 如有需要，增加 min_child_weight > 1
- selection metric：
  - 先按 validation Brier 排序为主；
  - Brier 差异很小时再用 AUC 作为 tie-breaker。

2. 校准策略
- 使用 train/val 做 Platt / isotonic calibration，选择在 val 上 Brier + ECE 最优的方案。
- calibration 仅基于 train+val，不泄漏 test。
- 在 test 上评估：
  - baseline（可同时跑一个 Platt 版本） vs v3（Platt）的 Brier / ECE / AUC。

3. 评估与集成
- 比较 logreg baseline vs xgboost-v2 vs xgboost-v3 在 test split 上的：
  - Brier、ECE、AUC、PR-AUC。
- 对 v3 也跑 edge-based settlement backtest（与 v2 相同配置），收集 net PnL + 风险指标。
- 更新 rerun 报告，将 v3 加入 Baseline vs Candidate 对比表，并给出明确结论：
  - 若 v3 test Brier < baseline 且 backtest 不弱于 v2，标记为“首选 challenger”。

验收标准
- [ ] train_xgboost_v3 可在同一训练集上跑通
- [ ] v3 的 test Brier 明确优于 baseline（数值上有安全 margin，不只是 0.000x 差距）
- [ ] rerun 报告包含 v3 的离线和回测结果，并对 v2/v3 进行清晰对比

**Comment by phead198708** [comment]
## Issue 50 rerun report: `xgboost-v3` challenger

Implemented and evaluated `xgboost-v3` on branch `codex/issue-50-xgboost-v3`.

### Bottom line

`xgboost-v3` **does solve the probability-quality problem** from this issue. It is now the best probability challenger by test Brier/ECE/AUC/PR-AUC. It should **not** be promoted as the first champion yet because production readiness and risk metrics are still incomplete, and v2 still has stronger edge-backtest utility in the current sweep.

Bootstrap decision for v3: **CONTINUE_BOOTSTRAP_EXPERIMENTATION**  
Confidence: **MEDIUM**

### Offline test metrics

| Model | Variant | Brier | ECE | AUC | PR AUC | Accuracy |
|---|---|---:|---:|---:|---:|---:|
| `logreg-baseline-v1` | raw | 0.2012 | 0.1436 | 0.6287 | 0.8372 | 0.7322 |
| `logreg-baseline-v1` | Platt | 0.1978 | 0.1318 | 0.6287 | 0.8372 | 0.7322 |
| `xgboost-v2` | raw | 0.2033 | 0.1985 | 0.7582 | 0.8766 | 0.6562 |
| `xgboost-v2` | Platt | 0.2278 | 0.2274 | 0.7582 | 0.8766 | 0.6758 |
| `xgboost-v3` | raw | 0.1331 | 0.1304 | 0.8503 | 0.9223 | 0.7979 |
| `xgboost-v3` | Platt | 0.1235 | 0.0927 | 0.8503 | 0.9223 | 0.8379 |

Key readout:
- Baseline Platt test Brier: **0.1978**
- v2 Platt test Brier: **0.2278**
- v3 Platt test Brier: **0.1235**
- v3 Platt test ECE: **0.0927**

### Selected v3 params

```json
{
  "colsample_bytree": 1.0,
  "eta": 0.05,
  "eval_metric": "logloss",
  "gamma": 0.0,
  "lambda": 5.0,
  "max_depth": 4,
  "min_child_weight": 2.0,
  "nthread": 1,
  "objective": "binary:logistic",
  "rounds": 200,
  "seed": 0,
  "subsample": 0.8,
  "tree_method": "hist"
}
```

### Edge settlement backtest, 5bps fee + 5bps slippage

Standard threshold grid:

| Model | Best Edge | Net PnL | Trades | Win Rate | Avg Net Return |
|---|---:|---:|---:|---:|---:|
| `logreg-baseline-v1` | 0.10 | 4.2893 | 21 | 71.4% | 40.1% |
| `xgboost-v2` | 0.10 | 8.2633 | 72 | 62.5% | 22.5% |
| `xgboost-v3` | 0.10 | 6.5636 | 91 | 58.2% | 14.2% |

Extended threshold scan through edge 0.30:

| Model | Best Edge | Net PnL | Trades | Win Rate | Avg Net Return |
|---|---:|---:|---:|---:|---:|
| `xgboost-v2` | 0.30 | 11.8369 | 65 | 69.2% | 35.7% |
| `xgboost-v3` | 0.30 | 8.7738 | 71 | 63.4% | 24.3% |

### Acceptance status

| Acceptance item | Status | Evidence |
|---|---|---|
| Add/run `train_xgboost_v3` on the same training set | PASS | `artifacts/models/xgboost-v3/*` generated |
| v3 test Brier clearly beats baseline | PASS | v3 Platt Brier 0.1235 vs baseline Platt 0.1978 |
| Include Brier/ECE/AUC/PR-AUC comparison | PASS | comparison artifact generated |
| Include edge settlement backtest | PASS | standard + extended threshold sweeps generated |
| Mark v3 preferred challenger only if backtest is not weaker than v2 | NOT MET | v3 improves probability quality but v2 has stronger current utility |

### Remaining blockers before champion promotion

- Backtest max drawdown missing
- Backtest Sharpe/Sortino missing
- Serving latency/error readiness report missing
- Model complexity notes missing

### Validation

- `388 passed, 2 skipped`
- `ruff check .` passed

Full local report artifact:
`/Users/tcscoder/Workspaces/BiGan/data/model-train-backtest-rerun-20260520T134420Z/artifacts/rerun_report_issue_50_xgboost_v3.md`
